File: parse_string.py
def parse_string(s: str) -> dict:
    res = {}
    list_s = s.split(";")

    for item in list_s:
        k, v = item.split("=")
        res[k] = v

    return res


print(parse_string("name=John;age=25;city=San Diego"))

File: parse_string.ts
type ObjKey = Record<string, string> 

const parse_string = (s: string): ObjKey => {
    const res: ObjKey = {}
    for (const item of s.split(";")) {
        const [k, v] = item.split("=");
        if (k !== undefined && v !== undefined) {
            res[k] = v;
        }
    }

    return res;
}

console.log(parse_string("name=John;age=25;city=San Diego"))